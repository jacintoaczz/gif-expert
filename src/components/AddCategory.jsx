import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  // Deberiamos darle un valor inicial al state, dado que, de no hacerlo, el state
  // tendra el valor de 'undefined', y esto nos maracara un error en la consola.
  const [inputValue, setInputValue] = useState("Ingrese una busqueda...");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <input
          type="text"
          onChange={handleInputChange}
          autoFocus={true}
          required={true}
          placeholder={" "}
        />
        <label htmlFor="input" className="control-label">
          Categor√≠a
        </label>
        <i className="bar"></i>
      </div>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
