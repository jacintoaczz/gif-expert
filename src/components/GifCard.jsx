import React from "react";

export const GifCard = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInLeft">
      <img src={url} alt={`${title}`} className="gif" />
      <section className="card__body">
        <h4>{title.toUpperCase()}</h4>
      </section>
    </div>
  );
};
