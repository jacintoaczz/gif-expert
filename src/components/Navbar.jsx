import React from "react";

export const Navbar = () => {
  return (
    <header className="navbar">
      <h1>Gif Expert</h1>
    </header>
  );
};
