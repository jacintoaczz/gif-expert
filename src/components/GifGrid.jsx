import React, { useEffect, useState } from "react";
import { GifCard } from "./GifCard";

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  // Si no le aÃ±adimos dependencias al useEffect (es decir, el ultimo argumento como '[]')
  // solo se ejecutara el efecto una sola vez. Cuando se crea el componente.
  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const searchUrl = `https://api.giphy.com/v1/gifs/search?q=${encodeURI(
      category
    )}&api_key=Glm6NFYw1SuuZ7eFCqoSzlEmlpTjNiEc&limit=5`;

    const response = await fetch(searchUrl);
    const { data } = await response.json();

    // Le damos formato a la informacion que llega
    const gifs = data.map((image) => {
      return {
        id: image.id,
        title: image.title,
        url: image.images?.downsized_medium.url,
      };
    });

    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      <div className="gifs-container">
        {images.map((image) => {
          // Usando el {...image} es otra forma de enviar props.
          return <GifCard key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
