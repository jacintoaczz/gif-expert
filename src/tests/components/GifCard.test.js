import { shallow } from "enzyme";
import { GifCard } from "../../components/GifCard";

describe("Pruebas en el </Gifcard />", () => {
  // Props del componente
  const id = "elId";
  const title = "Un titulo";
  const url = "https://localhost:3000/una-imagen";
  const wrapper = shallow(<GifCard id={id} title={title} url={url} />);

  test("Debe de mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de tener un h4 con el "title"', () => {
    const p = wrapper.find("h4");
    expect(p.text().trim()).toBe(title.toUpperCase());
  });

  test('Debe de tener una imagencon el "url", y con propiedad "alt" de los props.', () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test('Debe tener las clases para animar "animate__animated animate__fadeIn"', () => {
    const cardDiv = wrapper.find("div.card");
    const className = cardDiv.prop("className");
    expect(className.includes("animate__fadeInLeft")).toBe(true);
  });
});
