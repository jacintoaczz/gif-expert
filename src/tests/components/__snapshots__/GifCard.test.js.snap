// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el </Gifcard /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="Un titulo"
    className="gif"
    src="https://localhost:3000/una-imagen"
  />
  <section
    className="card__body"
  >
    <h4>
      UN TITULO
    </h4>
  </section>
</div>
`;
